components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java

```diff
@@ -230,25 +218,32 @@ private void insertAccessToken(String accessToken, String consumerKey, AccessTok
             } else {
                 insertTokenPrepStmt.setString(18, NONE);
             }
-            if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
-                if (OAuth2ServiceComponentHolder.isConsentedTokenColumnEnabled()) {
+            if (OAuth2ServiceComponentHolder.isConsentedTokenColumnEnabled()) {
+                if (OAuth2ServiceComponentHolder.isAcrColumnEnabled()) {
```

Comment:
Since this is introduced as a new feature, should we have flags like this.

Only this we need to make sure it the schema is modified when a new version is deployed anywhere

Reply:
Seems we need to provide backward compatibility support (The new table should work with the old binary and the new binary should work with the old DB schema) we have to introduce flags like this. Once the migration is completed, we can remove these IF conditions. We should create a separate issue to track this